### PR TITLE
Balance+stripe

### PR DESCRIPTION
**Pull request by abhishekchohan**
# What it does ?
- Stripe API implemented ( FE -BE )
- Balance Page
- Api calls to BE for fetching user data
- BE route to /charge added for handling stripe payment form
- Custom split card element design for stripe

# What to avoid ?
- N/A

# what to look for ?
- any code organisation, bugs, etc.
- Refactoring still needed a little bit here and there..

# Screenshots:

image

image

image

### CODE REVIEW

File: server/resources/stripe_checkout.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    @jwt_required
+    def post(self):
+        try:
+            data = request.get_json()
+            charge = stripe.Charge.create(
+            amount=data['credits']*1000,
+            currency='cad',
+            description='Credits Purchased',
+            source=data['token']['id']
+            )
```
Comment (ko1keith) [inline server/resources/stripe_checkout.py:21]: formatting, tab line 17-20

File: server/resources/stripe_checkout.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            if charge:
+                balance = UserModel.get_balance(get_jwt_identity())
+                balance = balance + data['credits']
+                UserModel.update_balance(get_jwt_identity(), balance)
+                # When I am sending a object back, I am getting errors
+                return "ok", 200
```
Comment (ko1keith) [inline server/resources/stripe_checkout.py]: I usually return json format like so -->  { "message": "ok"}, 200
Or if its and error  --> {"error": "something went wrong"}, 400

File: server/resources/stripe_checkout.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            else:
+                return "something was wrong with the transaction"
+        except Exception as e:
+            return "error: something went wrong"
```
Comment (ko1keith) [inline server/resources/stripe_checkout.py]: same as above

File: server/resources/stripe_checkout.py (hunk 1 above)
Comment (abhishekchohan) [reply server/resources/stripe_checkout.py:21]: I am confused with this formatting. Would you please give me a quick formatting sample ?

File: server/resources/stripe_checkout.py (hunk 2 above)
Comment (abhishekchohan) [reply server/resources/stripe_checkout.py]: sure..I will implement that in my next commit.. Thanks (I tried it before like this but got random errors dunno why.)
Thanks for the suggestion though.

File: server/resources/stripe_checkout.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+from flask import request
+from flask_restful import Resource
+from flask_jwt_extended import (
+    create_access_token, jwt_required, get_jwt_identity)
+from models.user_model import UserModel
+import stripe, os
+
+
+stripe.api_key = os.environ.get('STRIPE_SECRET_KEY')
```
Comment (ko1keith) [inline server/resources/stripe_checkout.py:9]: Maybe move this to the extensions.py file?

**Review by ko1keith — APPROVED** [review]
Other then some small formatting comments, everything else seems good! Might want to mention we need to add a new environment variable to our local .envs too

**Review by skassam21 — COMMENTED** [review]
I think @ko1keith means using prettier with Python on the back-end to improve the formatting of the code. 

I noticed that we used the old Stripe API - is there a reason you decided to use that vs. the newer one? I'm fine with it and I think this can be merged, but was wondering why

**Comment by abhishekchohan** [comment]
> I think @ko1keith means using prettier with Python on the back-end to improve the formatting of the code.
> 
> I noticed that we used the old Stripe API - is there a reason you decided to use that vs. the newer one? I'm fine with it and I think this can be merged, but was wondering why

- Ok thank you for clarification. I am actually using prettier but I don't know (python need a different one I guess)

- No there is no specific reason I guess. I tried all prebuild examples on documentation before and I was not able to get the same UI results as we have now (not even closer). Then I saw this example which seems to do the job.

**Review by skassam21 — APPROVED** [review]

**Pull request merged**